### DISCUSSION TITLE
Maintainer

### DISCUSSION THREAD

**Discussion by jphenow** [saml-idp/saml_idp#178] [General]
Some background; I started this project while working at a company from 2012-2015. It filled an immediate need we had, and I was happy to fill a community gap. I've been very far from SAML, and have sporadically been far from Ruby for the last 7 years. A handful of years ago someone came to me with really specific interest in making this its own GitHub org, taking on some maintainer roles, shepherding the library a little further. This sounded great except that person basically took over a bunch of "owner" roles then dropped a bunch of comments in issues then ghosted after about 72h. This soured me a bit on things around this project, maintaining etc. which isn't fair to the folks using and trying to contribute to it. 

It's a decent amount of time and effort to maintain and triage issues, but it still seems to scratch an itch for folks so I'd like to see the project continue on in whatever capacity it can as long as it is useful for folks.

All that said, @Zogoo, you've been the most regular and seem to have reasonable thoughts, expectations on the library. I'd like to retain some ownership of the org, library because a part of me still thinks I'll be able to help grow this a bit, but would you be interested in taking on some further maintainership of this project? At a minimum, I think this project needs someone who's a little more able than I to devote some time to helping folks get stable changes merged, release new gem versions.

Thoughts?

I was just going to flip you to maintainer and maybe see if we can either better automate gem release or just give you gem write access but then I realized I should probably say something first and see if you're _interested_ 😄

**Comment by Zogoo** [comment]
@jphenow  it's an honor for me. 
I like the structure of the gem code and how it's helped a lot of developers. Somehow, I connected to the authentication (IdP) area of development. Probably, it seems like I will continue work on authentications.

Before I put my answer. I would like to be honest from the beginning. And if you agree with following points I would like to try my best as owner of this gem.

Currently in my head I'm thinking that we need to grow the community. For that we might need to focus on another gem that uses this gem and works with devise and rails.

If you give me the owner role, l will follow your main guide and plus I would like to make the following changes.
1. To make this gem focused only SAML 2.0 protocol's request, a response handler (validation) and generator for SLO, SSO. For that we need to make it very simplify like only support one SP metadata model (which can easily serializeable). Drop extra logics like metadata downloader, extra dynamic logic supports for metadata persister and lambda executions. Instead of those logics we can just allow simple formats of data, for example:  array, hash. Also, if the gem just generates a response or handle a request, we should allow to users to control their own SP provider list (with config) instead of `service_providers` lambda selector.
2. Based on the above changes, we need to add new gem that works with devise or directly with rails.
So, if you think that I'm okay (as a side project) for an owner, I will try my best. 

Thanks
Zogoo

**Reply by jphenow** [reply]
Hey, I think that makes sense. I would probably encourage us to avoid _super_ deep rails/devise integration but I think it's fine to prioritize it as an entrypoint - does that make sense? Basically it seems to me there's a "core" piece, then there are pieces that touch integration points that make rails and devise work really well for this. 

I say this because it might make sense for folks to be able to pick and choose some parts of how to use the library, but I think its great if we can provide a smarter "happy path" for the general case. It also tends to be easier if we separate these concerns a bit.

I reread your last statement about having it in another gem that is _for_ the integration to devise/rails - yes I think that makes sense. Another layer of abstraction on top of this sounds great.

Agreed on moving to simpler configuration by the way - I'm positive why it was that way in the first place. I believe when I wrote it we had specific loading needs that may no longer apply (I'm not sure they still use this gem). In that vein though, we should be mindful of breaking changes and consider cutting a new Major version if we decide to change things like configuration expectations or public APIs.

Love these ideas!

> Before I put my answer

Let me know what you're thinking when you decide. We can keep conversations about how we want to manage this in discussions for now. If we need to spin up a more live chat like Discord we can consider that too.

Once you let me know whether you'd like to do this, I will elevate your permissions in GitHub, RubyGems and we can go from there.

Cheers

**Reply by Zogoo** [reply]
@jphenow it seems like my last version comment (fixed version some syntax and some grammar) is not published. Also, my last sentences that were saying that if you agreed those my points. I'm happy to take your offer.

About chat, let me create discord room on this week and invite the guys.

Thanks and Regards

**Reply by jphenow** [reply]
I've added you as a maintainer on the repo. We can figure out either automating or giving you more access on rubygems soon.

Cheers

**Reply by Zogoo** [reply]
After a long busy journey (moving to another country), I had some long holidays and I looked at the code again.
I have created a DISCORD group for us.
Please join with this link https://discord.gg/hSwjtCrT

**Reply by Zogoo** [reply]
@jphenow, @mjobin-mdsol could you guys please join the discord? Once you guys join the server let me change our official message channel to discord over the documents. Because sometimes peoples asks questions about how to configure over gitter chat I thought it might be better to use one channel only.

**Reply by jphenow** [reply]
I will join soon. Juggling some big stuff at my day-job and a new kid. I am unlikely to be super active there, but I'll do my best to not block progress on the project. 

Personally I'm fine if you want to update the docs to point at the discord now.

**Reply by mjobin-mdsol** [reply]
I joined, although, I am not super active on Discord
